server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java

```diff
@@ -59,6 +60,7 @@ public class TranslogTransferManager {
     private final BlobPath remoteMetadataTransferPath;
     private final BlobPath remoteBaseTransferPath;
     private final FileTransferTracker fileTransferTracker;
+    private final RemoteTranslogTransferTracker remoteTranslogTransferTracker;
```

Comment:
should we also be considering adding upload time for uploading the metadata file?

Reply:
This is being done in `org.opensearch.index.translog.RemoteFsTranslog#download`:

```java
...
TranslogTransferMetadata translogMetadata = translogTransferManager.readMetadata();
        if (translogMetadata != null) {
            long durationInMillis = (System.nanoTime() - downloadStartTime) / 1_000_000L;
            statsTracker.addDownloadTimeInMillis(durationInMillis);
...
```

Are you referring to something else or recommending that we add this as a distinct stat field?

Reply:
I am referring to upload time in upload path.